### PR TITLE
[IMP] XLSX: export text vertical alignment

### PR DESCRIPTION
**Pull request by yunchen4**
## Description:

This PR implements the functionality of exporting text vertical alignment.

Odoo task ID : [3249032](https://www.odoo.com/web#id=3249032&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/xlsx/helpers/content_helpers.ts [hunk 1]
```diff
@@ -129,7 +129,10 @@ export function extractStyle(cell: CellData, data: WorkbookData): ExtractedStyle
     numFmt: format ? { format: format, id: 0 /* id not used for export */ } : undefined,
     border: exportedBorder || {},
     alignment: {
-      vertical: "center" as XLSXVerticalAlignment, // we always center vertically for now
+      vertical:
+        style.verticalAlign === "middle"
+          ? "center"
+          : (style.verticalAlign as XLSXVerticalAlignment),
```
Comment (hokolomopo) [inline src/xlsx/helpers/content_helpers.ts:135]: Probably better to use the same convertion_map than at the import

**Review by hokolomopo — COMMENTED** [review]
👍

**Comment by LucasLefevre** [comment]
robodoo delegate=hokolomopo

File: tests/__snapshots__/xlsx_export.test.ts.snap [hunk 2]
```diff
@@ -10439,57 +10454,63 @@ Object {
...
     </borders>
-    <cellXfs count=\\"17\\">
+    <cellXfs count=\\"19\\">
         <xf numFmtId=\\"0\\" fillId=\\"0\\" fontId=\\"0\\" borderId=\\"0\\">
-            <alignment vertical=\\"center\\"/>
+            <alignment vertical=\\"bottom\\"/>
         </xf>
         <xf numFmtId=\\"0\\" fillId=\\"0\\" fontId=\\"1\\" borderId=\\"0\\">
-            <alignment vertical=\\"center\\"/>
+            <alignment/>
```
Comment (hokolomopo) [inline tests/__snapshots__/xlsx_export.test.ts.snap:10462]: Not a real issue, but we shouldn't put empty tags in the xlsx, it's useless

**Review by hokolomopo — COMMENTED** [review]
Just a small comment, otherwise it's good 👍

File: tests/__snapshots__/xlsx_export.test.ts.snap (hunk 2 above)
Comment (yunchen4) [reply tests/__snapshots__/xlsx_export.test.ts.snap]: @hokolomopo Changes done. I also made `xf` tag self-closed if there's no alignment.

**Comment by hokolomopo** [comment]
@yunchen4 can you rebase/remove import commit ? I changed the target of the branch to master :)

**Comment by yunchen4** [comment]
@hokolomopo Rebase done :)

**Comment by rrahir** [comment]
robodoo delegate=hokolomopo

**Comment by hokolomopo** [comment]
@robodoo r+

**Pull request closed without merging**